src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -30,9 +32,9 @@ internal void WriteAsV7(Stream archiveStream, Span<byte> buffer)
             long actualLength = GetTotalDataBytesToWrite();
             TarEntryType actualEntryType = GetCorrectTypeFlagForFormat(TarEntryFormat.V7);
 
-            int checksum = WriteName(buffer, out _);
+            int checksum = WriteNameSpan(buffer, out _);
             checksum += WriteCommonFields(buffer, actualLength, actualEntryType);
-            WriteChecksum(checksum, buffer);
+            _checksum = WriteChecksum(checksum, buffer);
```

Comment:
why the sync method writes the checksum to a field and the async method does not?

Reply:
TarHeader is a struct. It owns the _checksum field. In the sync method, I have no problem updating the value of _checksum inside the method: the target field gets updated. But in the async method, the TarHeader is copied into the method, so the _checksum value that gets updated is the one from the copied header, not the original one.

As discussed on chat, this is a good candidate for refactoring and simplification on a later PR.